LanguageFeatures/Extension-types/dynamic_semantics_extension_type_A06_t01.dart

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Let R0 be the extension type erasure of V0. At run time, the type
+/// V0 has the same representation and semantics as R0. In particular, they
+/// behave identically with respect to is, is!, as, and ==, both when V0 and R0
+/// are used as types, and when they occur as subterms of another type.
+///
+/// @description Check that an extension type erasure is not performed
+/// statically outside of switch expressions and statements
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=inline-class
+
+extension type BoolET1(bool _) {}
+extension type BoolET2(bool _) implements bool {}
+extension type const ConstBoolET1(bool _) {}
+extension type const ConstBoolET2(bool _) implements bool {}
+
+main() {
+  var _false = BoolET1(false);
+  var _true = BoolET2(true);
+  const True = ConstBoolET1(true);
+  const False = ConstBoolET2(false);

```

Comment:
Perhaps use standard conventions for naming (we aren't testing non-standard names):

```suggestion
var _false = BoolET1(false);
var _true = BoolET2(true);
const cTrue = ConstBoolET1(true);
const cFalse = ConstBoolET2(false);

main() {
```

This makes `_false` and `_true` standard private names, and it makes `cTrue` and `cFalse` standard camel-case names. I think this is good for the overall readability, and surely we have some other tests verifying that local variables can have "looks just like private" names, and non-types can have capitalized names, even though they shouldn't. ;-)

Reply:
(I had no other comments, so I went ahead and did this.)